### PR TITLE
Switched to the more performant System.Threading.Lock on .NET 9.0+

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**

### CODE REVIEW

File: src/Basic.CompilerLog.UnitTests/Basic.CompilerLog.UnitTests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net8.0;net472</TargetFrameworks>
+    <TargetFrameworks>net8.0;net9.0</TargetFrameworks>
```
Comment (jaredpar) [inline src/Basic.CompilerLog.UnitTests/Basic.CompilerLog.UnitTests.csproj:4]: Can't remove `net472` here as that is a TFM that I support and test.

File: src/Basic.CompilerLog.Util/Basic.CompilerLog.Util.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net8.0;net9.0;net472;netstandard2.0</TargetFrameworks>
```
Comment (jaredpar) [inline src/Basic.CompilerLog.Util/Basic.CompilerLog.Util.csproj:4]: This is removing platforms that I actively support.

**Review by jaredpar — CHANGES_REQUESTED** [review]
Not sure if I want to go this direction. The existing use of locking is not a known source of performance issues with this tool or library (the bottlenecks are very much elsewhere). Further taking this change relies on my removing several TFMs that I actively support today.

**Comment by MarkCiliaVincenti** [comment]
> Not sure if I want to go this direction. The existing use of locking is not a known source of performance issues with this tool or library (the bottlenecks are very much elsewhere). Further taking this change relies on my removing several TFMs that I actively support today.

I can fix the issues so that the factory method is used instead, but not sure if you still want this.
https://github.com/MarkCiliaVincenti/Backport.System.Threading.Lock?tab=readme-ov-file#factory-method-if-targeting-frameworks-prior-to-net-50

**Comment by jaredpar** [comment]
Thanks for the contribution but don't think I want to go in this direction. The performance around the limited locking here is not enough of a concern to introduce this disruptive of a change.

**Comment by MarkCiliaVincenti** [comment]
OK @jaredpar

**Comment by MarkCiliaVincenti** [comment]
@jaredpar a new version of Backport.System.Threading.Lock has come out that acts as a source generator and basically dropping the DLL as a dependency. If this new development makes you reconsider, I will amend this PR accordingly.

**Pull request closed without merging**